plugins/bookmarks/__init__.py

```diff
@@ -244,84 +230,73 @@ def delete_bookmark(self, widget, targets):
 
         self.save_db()
 
+    @common.idle_add()
```

Comment:
idle_add only ensures that something runs on the main UI thread -- so in fact by using it you will ensure that the main thread is blocked. :)

The GIO async stuff requires you use the _async version of whatever calls that exist.

`@threaded` is what you want here for now. Using the GIO async stuff is a bit awkward, and to do it well we really need to wrap it with coroutines of some kind...